model/ingress.go

```diff
@@ -95,3 +98,24 @@ func (ia *IngressAnnotations) SetDefaults() {
 func (ia *IngressAnnotations) SetHibernatingDefaults() {
 	ia.ConfigurationSnippet = "return 410;"
 }
+
+func ConfigureIngressAnnotations(whitelist []string, existingSnippet string) *IngressAnnotations {
+	// Create the allow directives for each IP in the whitelist
+	var allowDirectivesBuilder strings.Builder
+	for _, ip := range whitelist {
+		if ip != "" {
+			allowDirectivesBuilder.WriteString(fmt.Sprintf("allow %s;\n", ip))
+		}
+	}
+
+	configSnippet := existingSnippet + "\n" + allowDirectivesBuilder.String() + `
+       deny all;
+       error_page 403 =419 /custom_419_page;
+   `
+
+	// Setting up the IngressAnnotations struct
+	ia := &IngressAnnotations{
+		ConfigurationSnippet: configSnippet,
+	}
+	return ia
```

Comment:
I think you could continue to use strings.Builder to make it a little easier on the eyes. Something like:

```suggestion
    // Build the configSnippet using strings.Builder
    configSnippetBuilder := strings.Builder{}
    configSnippetBuilder.WriteString(existingSnippet)
    configSnippetBuilder.WriteString("\n")
    configSnippetBuilder.WriteString(allowDirectivesBuilder.String())
    configSnippetBuilder.WriteString(`
       deny all;
       error_page 403 =419 /custom_419_page;
   `)
    configSnippet := configSnippetBuilder.String()

    // Setting up the IngressAnnotations struct
    ia := &IngressAnnotations{
        ConfigurationSnippet: configSnippet,
    }
    return ia
```